pkg/util/limiter/utilization.go

```diff
@@ -29,21 +33,55 @@ const (
 )
 
 type utilizationScanner interface {
-	// Scan returns CPU time in seconds and memory utilization in bytes, or an error.
-	Scan() (float64, uint64, error)
+	// Scan returns procfs CPU time and cgroup CPU time in seconds and memory utilization in bytes, or an error.
+	Scan() (float64, float64, uint64, error)
 }
 
 type procfsScanner struct {
 	proc procfs.Proc
 }
 
-func (s procfsScanner) Scan() (float64, uint64, error) {
+func scanText(path string) (string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to open %s", path)
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	if !scanner.Scan() {
+		return "", fmt.Errorf("unable to scan %s", path)
+	}
+	if scanner.Err() != nil {
+		return "", errors.Wrapf(err, "unable to scan %s", path)
+	}
+
+	return scanner.Text(), nil
+}
+
+func (s procfsScanner) Scan() (float64, float64, uint64, error) {
 	ps, err := s.proc.Stat()
 	if err != nil {
-		return 0, 0, errors.Wrap(err, "failed to get process stats")
+		return 0, 0, 0, errors.Wrap(err, "failed to get process stats")
+	}
+
+	const cgrpV1Path = "/sys/fs/cgroup/cpu/cpuacct.usage"
+	cgrpCPUStr, err := scanText(cgrpV1Path)
+	var cgrpCPU float64
+	if err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			return 0, 0, 0, err
+		}
```

Comment:
if `err != nil` and err is `fs.ErrNotExist` the code will flow to the return statement below. Is it expected?